lib/core/request.js

```diff
@@ -23,8 +23,8 @@ class Request {
     upgrade,
     requestTimeout
   }, handler) {
-    if (typeof path !== 'string' || path[0] !== '/') {
-      throw new InvalidArgumentError('path must be a valid path')
+    if (typeof path !== 'string') {
```

Comment:
Could we have a revés that does a more strict check?

Reply:
Yep,
```
let isURL = false;
try {
  new URL(path);
  isURL = true;
} catch {}
console.log(isURL);
```

Reply:
```
if (typeof path !== 'string' || (path[0] !== '/' && isURL === false))
```